pkg/handlers/generic/lifecycle/addons/helmaddon.go

```diff
@@ -227,7 +227,10 @@ func waitToBeReady(
 			Reader: client,
 			Target: hcp.DeepCopy(),
 			Check: func(_ context.Context, obj *caaphv1.HelmChartProxy) (bool, error) {
-				return conditions.IsTrue(obj, caaphv1.HelmReleaseProxiesReadyCondition), nil
+				if obj.ObjectMeta.Generation != obj.Status.ObservedGeneration {
```

Comment:
I noticed this when moving the function. I think its a good change to prevent the function from returning early, especially during cluster upgrades where the HCP already has HelmReleaseReadyCondition but its out of date.